### ISSUE TITLE
training on gpu

### ISSUE THREAD

**Issue by islamtashfiq** [fvisin/main_loop_tf#18]
I got the memory issue 
OOM when allocating tensor with shape[1,60,60,1024]         [[Node: gpu0_train/my_model/resnet_v2_101/block3/unit_8/bottleneck_v2/preact/moments/SquaredDifference = SquaredDifference[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"](gpu0_train/my_model/resnet_v2_101/block3/unit_7/bottleneck_v2/add, gpu0_train/my_model/resnet_v2_101/block3/unit_8/bottleneck_v2/preact/moments/StopGradient)]]

Which is caused by training on cpu. I changed the devices to gpu but it is still on cpu. Also I changed the with tf.device('/cpu:0') line to with tf.device('/gpu:0'). But nothing changed. How can I make training on one gpu device.
Thanks

**Comment by fvisin** [comment]
As you can see from the stacktrace the node is actually allocated on gpu,
as it should.

This is not a bug of the main_loop but rather depends on your gpu card not
having enough free memory to allocate the model. This can depend on a
variety of factors but I am afraid their analysis and solution transcends
the kind of support we can offer.

Il 02 ago 2017 4:48 PM, "islamtashfiq" <notifications@github.com> ha
scritto:

**Comment by islamtashfiq** [comment]
Thanks for your reply.
My GPU is TITAN X with 12 GB free memory. I cropped images to size [473,473] and batch size 1. My question is about this line that says _device="/job:localhost/replica:0/task:0/**cpu**:0"

**Comment by fvisin** [comment]
Part of the computation has to be done on cpu, e.g., loading the data and
filling the gpus as well as collecting the gradients from the gpus.

There are several tutorials on how to write (multi)gpu code for tensorflow,
if you want to have a better grasp of the topic I strongly suggest you to
start from them and try writing your own simple code to have a feeling of
what's happening before moving to a more powerful yet admittedly more
complicated framework like ours. :)

Il 02 ago 2017 7:39 PM, "islamtashfiq" <notifications@github.com> ha
scritto:

**Comment by islamtashfiq** [comment]
I am sure that training is done on cpu as I monitor the gpu with nvidia-smi and it shows no process. Now it is in training but on cpu. Thanks anyway.

**Comment by islamtashfiq** [comment]
I found my mistake :).
